### PR TITLE
Add PSQL ODBC tests for XML datatype

### PR DESCRIPTION
**Pull request by rossd-bitquill**
### Description

This PR adds tests for the PSQL XML datatype including testing:
Table Creation
Insertion Success
Update Success
View Creation

Signed-off-by: rossd-bitquill <dutkiert@amazon.com>

### Issues Resolved
BABELFISH-563

### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/odbc/psqlodbc/xml.cpp [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+TEST_F(PSQL_DataTypes_Xml, Insertion_Success) {
+  vector<string> inserted_values = {
+    "NULL",
+    SIMPLE_XML,
+    NORMAL_XML,
+    LARGE_XML,
+  };
+  const int number_of_entries = inserted_values.size();
```
Comment (ghost) [inline test/odbc/psqlodbc/xml.cpp]:
```suggestion
  const int NUMBER_OF_ENTRIES = inserted_values.size();
```

File: test/odbc/psqlodbc/xml.cpp [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+TEST_F(PSQL_DataTypes_Xml, Data_Length) {
+  vector<string> inserted_values = {
+    "NULL",
+    SIMPLE_XML,
+    NORMAL_XML,
+    LARGE_XML
+  };
+  const int number_of_entries = inserted_values.size();
```
Comment (ghost) [inline test/odbc/psqlodbc/xml.cpp]:
```suggestion
  const int NUMBER_OF_ENTRIES = inserted_values.size();
```

File: test/odbc/psqlodbc/xml.cpp [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+};
+
+const string SIMPLE_XML = "<root>\
+<book>-148708948</book>\
+<leaf>simple</leaf>\
+<remain>shake</remain>\
+<football>nest</football>\
+<blanket>1275892738</blanket>\
+<try>-249153758.16992378</try>\
+</root>";
```
Comment (ghost) [inline test/odbc/psqlodbc/xml.cpp]:
```suggestion
  <book>-148708948</book>\
  <leaf>simple</leaf>\
  <remain>shake</remain>\
  <football>nest</football>\
  <blanket>1275892738</blanket>\
  <try>-249153758.16992378</try>\
  </root>";
```

File: test/odbc/psqlodbc/xml.cpp [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+</stems>\
+<breeze>1269756220</breeze>\
+<cookies>544666062</cookies>\
+<yesterday>grade</yesterday>\
+<power>-663154896.7426605</power>\
+<topic>stomach</topic>\
+</song>\
+<fact>-1834471409</fact>\
+<repeat>-1980145795</repeat>\
+</root>";
```
Comment (ghost) [inline test/odbc/psqlodbc/xml.cpp]:
```suggestion
  <that>hot</that>\
  <anyone>1386369263.1890697</anyone>\
  <finest>believed</finest>\
  <song>\
  <stems>\
  <headed>1563641070.153514</headed>\
  <bring>-1175941901.956747</bring>\
  <stand>-494359546.04893684</stand>\
  <pole>774686962</pole>\
  <outer>production</outer>\
  <why>applied</why>\
  </stems>\
  <breeze>1269756220</breeze>\
  <cookies>544666062</cookies>\
  <yesterday>grade</yesterday>\
  <power>-663154896.7426605</power>\
  <topic>stomach</topic>\
  </song>\
  <fact>-1834471409</fact>\
  <repeat>-1980145795</repeat>\
  </root>";
```

File: test/odbc/psqlodbc/xml.cpp [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+const string LARGE_XML = "<Root>\
+ <sys.tables name=\"tnullfixeddecimal0100\" object_id=\"4195065\" schema_id=\"1\" parent_object_id=\"0\" type=\"U \" type_desc=\"USER_TABLE\" create_date=\"2022-02-03T19:23:37.030\" modify_date=\"2022-02-03T19:23:37.030\" is_ms_shipped=\"0\" is_published=\"0\" is_schema_published=\"0\" lob_data_space_id=\"0\" max_column_id_used=\"5\" lock_on_bulk_load=\"0\" uses_ansi_nulls=\"1\" is_replicated=\"0\" has_replication_filter=\"0\" is_merge_published=\"0\" is_sync_tran_subscribed=\"0\" has_unchecked_assembly_data=\"0\" text_in_row_limit=\"0\" large_value_types_out_of_row=\"0\" is_tracked_by_cdc=\"0\" lock_escalation=\"0\" lock_escalation_desc=\"TABLE\" is_filetable=\"0\" is_memory_optimized=\"0\" durability=\"0\" durability_desc=\"SCHEMA_AND_DATA\" temporal_type=\"0\" temporal_type_desc=\"NON_TEMPORAL_TABLE\" is_remote_data_archive_enabled=\"0\" is_external=\"0\" is_node=\"0\" is_edge=\"0\" />\
```
Comment (ghost) [inline test/odbc/psqlodbc/xml.cpp]: Nit: This is indented by 1 space. Can you fix indentation by indenting to 2 spaces?

File: test/odbc/psqlodbc/xml.cpp [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+  const vector<int> BBF_LENGTH_EXPECTED = {10, 0};
+  const vector<int> BBF_PRECISION_EXPECTED = {10, 0};
+  const vector<int> BBF_SCALE_EXPECTED = {0, 0};
+  const vector<string> BBF_NAME_EXPECTED = {"int", "xml"};
+
+  testCommonColumnAttributes(ServerType::MSSQL, BBF_TABLE_NAME, 
+      TABLE_COLUMNS.size(), COL1_NAME, 
+      BBF_LENGTH_EXPECTED, BBF_PRECISION_EXPECTED, 
+      BBF_SCALE_EXPECTED, BBF_NAME_EXPECTED);
```
Comment (ghost) [inline test/odbc/psqlodbc/xml.cpp]:
```suggestion
    TABLE_COLUMNS.size(), COL1_NAME, 
    BBF_LENGTH_EXPECTED, BBF_PRECISION_EXPECTED, 
    BBF_SCALE_EXPECTED, BBF_NAME_EXPECTED);
```

File: test/odbc/psqlodbc/xml.cpp [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+  createView(ServerType::MSSQL, BBF_VIEW_NAME, BBF_VIEW_QUERY);
+
+  verifyValuesInObject(ServerType::MSSQL, BBF_VIEW_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+  verifyValuesInObject(ServerType::PSQL, PG_VIEW_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+
+  dropObject(ServerType::MSSQL, "VIEW", BBF_VIEW_NAME);
+  dropObject(ServerType::PSQL, "VIEW", PG_VIEW_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+}
```
Comment (ghost) [inline test/odbc/psqlodbc/xml.cpp]: Nit: Please add new line at end of file.

Comment (rossd-bitquill) [reply test/odbc/psqlodbc/xml.cpp]: Done.

**Review by ghost — DISMISSED** [review]

File: test/odbc/psqlodbc/xml.cpp [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+  inserted_values = duplicateElements(inserted_values);
+
+  verifyValuesInObject(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, Data_Length) {
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/xml.cpp]: What's the difference between this test and `Insertion_Success`

File: test/odbc/psqlodbc/xml.cpp [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  testUpdateSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, COL2_NAME, data_update_values, data_update_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, data_update_values, data_update_values);
+
+  data_update_values = {
+    SIMPLE_XML
+  };
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/xml.cpp]:
```suggestion
  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, data_update_values, data_update_values);
```

File: test/odbc/psqlodbc/xml.cpp (hunk 8 above)
Comment (rossd-bitquill) [reply test/odbc/psqlodbc/xml.cpp]: Excellent question.  I had a different version of this at one point, but it wasn't working so I thought I removed it, but I guess I didn't.

Will remove it.

File: test/odbc/psqlodbc/xml.cpp (hunk 9 above)
Comment (rossd-bitquill) [reply test/odbc/psqlodbc/xml.cpp]: The whole purpose of putting something different was to test that the update actually happened with different data, but when I factored the XML text out into file-level constants I forgot that, so changed instead to LARGE_XML to have a different value update test.

File: test/odbc/psqlodbc/xml.cpp [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+  vector <string> data_update_values = {
+    SIMPLE_XML
+  };
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  testUpdateSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, COL2_NAME, data_update_values, data_update_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, data_update_values, data_update_values);
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/xml.cpp]: Sorry, didn't catch this earlier.

`testUpdateSuccess` already does something similar to `verifyValuesInObject`

`verifyValuesInObject` should be used on the other connection
ie After updating with PG connection, `verifyValuesInObject` on BBF

Comment (rossd-bitquill) [reply test/odbc/psqlodbc/xml.cpp]: Ah, that makes sense, so I'll switch the two calls to varifyValuesInObject to refer to the other connection type.

**Review by ColinKYuen — DISMISSED** [review]

File: test/odbc/psqlodbc/xml.cpp [hunk 11]
```diff
@@ -0,0 +1,207 @@
+#include "psqlodbc_tests_common.h"
+#include "../src/query_generator.h"
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/xml.cpp]:
```suggestion
```

**Review by leszekamz — APPROVED** [review]

**Pull request merged**